merge_multi_operator_test.go

```diff
@@ -0,0 +1,82 @@
+package gorocksdb
+
+import (
+	"testing"
+
+	"github.com/facebookgo/ensure"
+)
+
+func TestMergeMultiOperator(t *testing.T) {
+	var (
+		givenKey     = []byte("hello")
+		startingVal  = []byte("foo")
+		mergeVal1    = []byte("bar")
+		mergeVal2    = []byte("baz")
+		fMergeResult = []byte("foobarbaz")
+		pMergeResult = []byte("barbaz")
+	)
+
+	merger := &mockMergeMultiOperator{
+		fullMerge: func(key, existingValue []byte, operands [][]byte) ([]byte, bool) {
+			ensure.DeepEqual(&fatalAsError{t}, key, givenKey)
+			ensure.DeepEqual(&fatalAsError{t}, existingValue, startingVal)
+			ensure.DeepEqual(&fatalAsError{t}, operands[0], pMergeResult)
+			return fMergeResult, true
+		},
+		partialMerge: func(key, leftOperand, rightOperand []byte) ([]byte, bool) {
+			t.FailNow() // this should never be called
+			return nil, false
+		},
+		partialMergeMulti: func(key []byte, operands [][]byte) ([]byte, bool) {
```

Comment:
Can you make sure `partialMergeMulti` is actually called in the test? Maybe increase a counter on the call?

Reply:
Full merge checks for a value that is returned from partial.   The whole thing would fail without it....

Reply:
Cool, didn't realize that.